### ISSUE TITLE
no-use-before-declare doesn't handle class members

### ISSUE THREAD

**Issue by NN---** [palantir/tslint#3011]
### Bug Report

- __TSLint version__: 5.5.0
- __TypeScript version__: 2.4.1
- __Running TSLint via__: gulp

#### TypeScript code being linted

```ts
class A { b: B = new B(); }
class B { }
```

with `tslint.json` configuration:

```json
no-use-before-declare: true
```

#### Actual behavior
variable B is used before declaration

#### Expected behavior
No error

**Comment by Patrik-Lundqvist** [comment]
I also experience the same issue inside class members functions.

```ts
class A {
  test() {
      const b = new B(); // variable 'B' used before declaration
  }
}

class B {}
```

**Comment by ajafff** [comment]
This request is not valid. Classes are not hoisted and can therefore not be used before declaration.

The following code would fail at runtime:

```ts
class A { b: B = new B(); }
new A();
class B { }
```

The current behavior of the rule is correct.

**Comment by kodicat** [comment]
What about the code

```
export class One {
    toSecond() {
        return new Two();
    }
}

export class Two {
    toFirst() {
        return new One();
    }
}
```

It will not fail at runtime.
